orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/SpinnakerMetadataServerGroupTagGenerator.java

```diff
@@ -59,11 +73,66 @@
       value.put("comments", (String) context.get("comments"));
     }
 
+    String cluster = null;
+    try {
+      cluster = Names.parseName(serverGroup).getCluster();
+
+      Map<String, Object> previousServerGroup = getPreviousServerGroup(
+        execution.getApplication(),
+        account,
+        cluster,
+        cloudProvider,
+        location
+      );
+
+      if (previousServerGroup != null) {
+        value.put("previousServerGroup", previousServerGroup);
+      }
+    } catch (Exception e) {
+      // failure to populate `previousServerGroup` is not considered a fatal error that would cause this task to fail
+      log.error("Unable to determine ancestor image details for {}:{}:{}:{}", cloudProvider, account, location, cluster, e);
+    }
+
     Map<String, Object> tag = new HashMap<>();
     tag.put("name", "spinnaker:metadata");
     tag.put("value", value);
 
     return Collections.singletonList(tag);
   }
 
+  Map<String, Object> getPreviousServerGroup(String application,
+                                             String account,
+                                             String cluster,
+                                             String cloudProvider,
+                                             String location) {
+    return retrySupport.retry(() -> {
+      try {
+        Map<String, Object> targetServerGroup = oortService.getServerGroupSummary(
+          application,
+          account,
+          cluster,
+          cloudProvider,
+          location,
+          "ANCESTOR",
+          "image",
+          "true"
```

Comment:
I don't know what crazy person made this a `String` but I didn't take on changing it as part of this PR.

Reply:
I really hope it wasn't me!